spek-data-driven-extension/src/test/kotlin/org/jetbrains/spek/data_driven/WhereSpec.kt

```diff
@@ -0,0 +1,32 @@
+package org.jetbrains.spek.data_driven
+
+import org.jetbrains.spek.api.Spek
+import org.jetbrains.spek.api.dsl.describe
+import org.jetbrains.spek.api.dsl.it
+import kotlin.test.assertEquals
+
+/**
+ * @author Niels Falk
+ */
+object WhereSpec : Spek({
+    describe("analyze names") {
+        on("%s ", { name, expectedLen, expectPalindrome ->
+            it("has a length $expectedLen") {
+                assertEquals(expectedLen, name.length)
+            }
+            it("${expectPalindrome.isIsNot()} a palindrome") {
+                assertEquals(expectPalindrome, name.isPalindrome())
+            }
+        }, where<String,Int,Boolean> {
+            //@formatter:off
+            //"name"  / "expectedLen" / "expectPalindrome"
+            ! "Niels" / 5             / false
```

Comment:
yes I'd prefer | and || to separate expectations like Spock does. But Kotlin doesn't allow to overwrite these operators jet. Maybe I and II (letters not pipe) as infix function are better.